frontend/src/components/NavBar.vue

```diff
@@ -1,180 +1,188 @@
+        </a>
+      </div>
+      <div class="beta-bump navbar-menu" :class="{ 'is-active': showMenu }">
+        <div class="navbar-start">
+          <a class="navbar-item" @click.prevent="newList()">
+            <span class="icon is-large navbar-icon">
+              <i class="fas fa-lg fa-plus-circle"></i>
+            </span>
+            New
+          </a>
+          <router-link v-if="currentUser.loggedIn" class="navbar-item" to="/s/user">
+            <span class="icon is-large navbar-icon">
+              <i class="fas fa-lg fa-user-circle"></i>
+            </span>
+            My Lists
+          </router-link>
+          <a href="https://aka.ms/theurlist" class="navbar-item" target="_new">
+            <span class="icon is-large navbar-icon">
+              <i class="fas fa-lg fa-question-circle"></i>
+            </span>
+            About
+          </a>
+        </div>
+
+        <div class="navbar-end">
+          <div class="navbar-item has-dropdown is-hoverable" v-if="currentUser.loggedIn">
+            <a class="navbar-link">
+              <div class="columns is-gapless is-mobile">
+                <div class="column is-narrow">
+                  <figure id="profileImage" class="image">
+                    <img class="is-rounded" :src="currentUser.profileImage" alt />
+                  </figure>
+                </div>
+                <div class="column">
+                  <span>{{ currentUser.name }}</span>
+                </div>
+              </div>
+            </a>
+            <div class="navbar-dropdown">
+              <a class="navbar-item" @click.prevent="logout()">
+                <span class="icon is-medium navbar-icon">
+                  <i class="fas fa-sign-out-alt"></i>
+                </span>
+                Log Out
+              </a>
+            </div>
+          </div>
+          <a v-else class="navbar-item" @click.prevent="login()">
+            <span class="icon is-large navbar-icon">
+              <i class="fas fa-lg fa-sign-in-alt"></i>
+            </span>
+            Login
+          </a>
+        </div>
+      </div>
+    </nav>
+  </header>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+import ProgressBar from "@/components/ProgressBar.vue";
+import UserMenu from "@/components/UserMenu.vue";
+import { UserManager } from "oidc-client";
+import User from "@/models/User";
+import config from "@/config";
+
+@Component({
+  components: {
+    ProgressBar,
+    UserMenu
+  }
+})
+export default class extends Vue {
+  showMenu: boolean = false;
+  showLoginModal: boolean = false;
+  userManager: UserManager;
+
+  constructor() {
+    super();
+    this.userManager = new UserManager({
+      automaticSilentRenew: true,
+      loadUserInfo: false,
+      authority: `https://${config.openId.domain}.b2clogin.com/${config.openId.domain}.onmicrosoft.com/v2.0/.well-known/openid-configuration?p=${config.openId.policy}`,
```

Comment:
Is it possible to create some data model / controller that constructs all of these special strings oppose to doing them in the vue.

TBH, not sure what the best practice is with vue here.

Reply:
Updated PR description.  Also moved all this auth related stuff into a dedicated auth service.